app/src/main/java/net/geekinpurple/www/appofourown/SearchResults.java

```diff
@@ -54,38 +58,44 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private class Retrieval extends AsyncTask<String, Void, String> {
-        Document doc;
+    @Override
+    public void onListItemClick(ListView l, View v, int position, long id) {
+        Work message = (Work) l.getItemAtPosition(position);
+        Intent i = new Intent(this, WorkInstance.class);
+        i.putExtra("work", message);
+        Log.d("testing", message.toString());
+        startActivity(i);
+    }
 
+    private class Retrieval extends AsyncTask<String, Void, ArrayList<Work>> {
         @Override
-        protected String doInBackground(String... params) {
+        protected ArrayList<Work> doInBackground(String... params) {
             String url = params[0];
             try {
-                Document ao3 =  Jsoup.connect(url).get();
-                Elements works = ao3.getElementsByClass("blurb");
+                Document doc =  Jsoup.connect(url).get();
```

Comment:
good point.

ok, this neeeeeeeds a refactor, it's duplicated code at that point.